autoscaler-example.yaml

```diff
@@ -0,0 +1,73 @@
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: nginx-cluster-proportional-autoscaler-params
+  namespace: default
+data:
+  simple-scalar.scale-map: |-
+    { 
+      "cores_to_replicas_map":
+      [
+        [ 1,1 ],
+        [ 3,3 ],
+        [ 512,5 ],
+        [ 1024,7 ],
+        [ 2048,10 ],
+        [ 4096,15 ],
+        [ 8192,20 ],
+        [ 12288,30 ],
+        [ 16384,40 ],
+        [ 20480,50 ],
+        [ 24576,60 ],
+        [ 28672,70 ],
+        [ 32768,80 ],
+        [ 65535,100 ]
+      ],
+      "nodes_to_replicas_map":
+      [
+        [ 1,1 ],
+        [ 2,2 ]
+      ]
+    }
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: nginx-autoscale-example
+  namespace: default
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        run: nginx-autoscale-example
+    spec:
+      containers:
+      - name: nginx-autoscale-example
+        image: nginx
+        ports:
+        - containerPort: 80
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: cluster-proportional-autoscaler
+  namespace: default
+  labels:
+    version: v1
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        version: v1
+    spec:
+      containers:
+        - image: gcr.io/zihongz-kubernetes-codelab/cluster-proportional-autoscaler:0.6
+          name: autoscaler
+          command:
+            - /cluster-proportional-autoscaler
+            - --namespace=default
+            - --configmap=nginx-cluster-proportional-autoscaler-params
```

Comment:
Wouldn't it be easier to assume namespace is the same we run in, and the configmap is mounted inside the pod? Or does this way of passing the configmap name help with dynamic config updates?